Multiple Run Chunk option does not work as expected - Duplication of global.codecept_dir

#### What are you trying to achieve?

I am trying to run tests in chunks in parallel. 

#### What do you get instead?

> Cannot find module '/Users/sabriM/Projects/abc/e2e/Users/sabriM/Projects/abc/e2e/tests/header-test.js'

As you see, `/Users/sabriM/Projects/abc/e2e` is duplicated. 

Probably line 97 of `codeceptjs/lib/codecept.js` file causes that. `loadtests` function of Codecept class.
```js
loadTests(pattern) {
    pattern = pattern || this.config.tests;

    const options = {
      cwd: global.codecept_dir,
    };

    glob.sync(pattern, options).forEach((file) => {
     // seems following line adds an extra global.codecept_dir. 
      this.testFiles.push(fsPath.resolve(fsPath.join(global.codecept_dir, file))); 
    });
  }
``` 

### Details

* CodeceptJS version: 1.2
* NodeJS Version: v8.9.1
* Operating System: High Sierra v10.13.4
* Protractor || WebDriverIO || Nightmare version (if related) I am using puppeteer (1.3.0)
* Configuration file:

```js
    tests: './tests/**/*-test.js',
    grep: process.env.testTag || null,
    output: process.env.imageFolder || './output',
    multiple:{
        parallel:{
            chunks: 2,
            browsers: ["chrome"],
        }
    },
    helpers: {
        Puppeteer: {
            url: 'https://local.dev.abc.com',
            chrome: {
                ignoreHTTPSErrors: true,
                executablePath: process.env.chromePath || '/usr/bin/google-chrome',
                args: ['--no-sandbox', '--disable-setuid-sandbox'],
            },
            windowSize: '1920x1080',
            show: false,
            waitForTimeout: 12000,
            waitForAction: 1000
        },
include: {
        I: './steps-file.js',
        allCoursesTab: './pages/aaa/all-courses-tab.js',
        automaticMessagesPage: './pages/aaa/automatic-messages-page.js',
        basicsPage: './pages/aaa/basics-page.js',
        bulkFileUploaderPage: './pages/aaa/bulk-file-uploader-page.js',
....
},
mocha: {
        retries: process.env.retries || 0
    },
....
```

--- a/examples/codecept.json
+++ b/examples/codecept.json
@@ -3,9 +3,8 @@
   "timeout": 10000,
   "output": "./output",
   "helpers": {
-    "WebDriverIO": {
+    "Puppeteer": {
       "url": "http://localhost",
-      "browser": "chrome",
       "restart": true
     }
   },
@@ -15,6 +14,9 @@
     }
   },
   "multiple": {
+    "parallel": {
+      "chunks": 2
+    },
     "default": {
       "grep": "signin",
       "browsers": ["chrome", "firefox"]

--- a/examples/yahoo_test.js
+++ b/examples/yahoo_test.js
@@ -1,8 +1,7 @@
 Feature('Yahoo test');
 
 Scenario('Nightmare basic test', (I) => {
-  I.amOnPage('http://127.0.0.1:8000');
-  pause();
+  I.amOnPage('http://yahoo.com');
   I.fillField('p', 'github nightmare');
   I.click('Search Web');
   I.waitForElement('#main', 2);

--- a/lib/codecept.js
+++ b/lib/codecept.js
@@ -3,6 +3,7 @@ const readFileSync = require('fs').readFileSync;
 const Container = require('./container');
 const Config = require('./config');
 const event = require('../lib/event');
+const { detectAbsolutePath } = require('./utils');
 const glob = require('glob');
 const runHook = require('./hooks');
 
@@ -89,12 +90,11 @@ class Codecept {
   loadTests(pattern) {
     pattern = pattern || this.config.tests;
 
-    const options = {
-      cwd: global.codecept_dir,
-    };
-
-    glob.sync(pattern, options).forEach((file) => {
-      this.testFiles.push(fsPath.resolve(fsPath.join(global.codecept_dir, file)));
+    glob.sync(pattern, {}).forEach((file) => {
+      if (!detectAbsolutePath(file)) {
+        file = fsPath.join(global.codecept_dir, file);
+      }
+      this.testFiles.push(fsPath.resolve(file));
     });
   }
 

--- a/lib/codecept.js
+++ b/lib/codecept.js
@@ -89,7 +89,11 @@ class Codecept {
   loadTests(pattern) {
     pattern = pattern || this.config.tests;
 
-    glob.sync(pattern, {}).forEach((file) => {
+    const options = {
+      cwd: global.codecept_dir,
+    };
+
+    glob.sync(pattern, options).forEach((file) => {
       if (!fsPath.isAbsolute(file)) {
         file = fsPath.join(global.codecept_dir, file);
       }

--- a/lib/codecept.js
+++ b/lib/codecept.js
@@ -3,7 +3,6 @@ const readFileSync = require('fs').readFileSync;
 const Container = require('./container');
 const Config = require('./config');
 const event = require('../lib/event');
-const { detectAbsolutePath } = require('./utils');
 const glob = require('glob');
 const runHook = require('./hooks');
 
@@ -91,7 +90,7 @@ class Codecept {
     pattern = pattern || this.config.tests;
 
     glob.sync(pattern, {}).forEach((file) => {
-      if (!detectAbsolutePath(file)) {
+      if (!fsPath.isAbsolute(file)) {
         file = fsPath.join(global.codecept_dir, file);
       }
       this.testFiles.push(fsPath.resolve(file));

--- a/lib/command/run-multiple.js
+++ b/lib/command/run-multiple.js
@@ -54,6 +54,7 @@ module.exports = function (selectedRuns, options) {
   const done = () => event.emit(event.multiple.before, null);
   runHook(config.bootstrapAll, done, 'multiple.bootstrap');
 
+
   const childProcessesPromise = new Promise((resolve, reject) => {
     processesDone = resolve;
   });
@@ -64,6 +65,10 @@ module.exports = function (selectedRuns, options) {
     const runConfig = run.getConfig();
     runsToExecute.push(executeRun(runName, runConfig));
   });
+
+  if (!runsToExecute.length) {
+    fail('Nothing scheduled for execution');
+  }
 
   // Execute all forks
   totalSubprocessCount = runsToExecute.length;

--- a/lib/command/run-multiple.js
+++ b/lib/command/run-multiple.js
@@ -54,6 +54,10 @@ module.exports = function (selectedRuns, options) {
   const done = () => event.emit(event.multiple.before, null);
   runHook(config.bootstrapAll, done, 'multiple.bootstrap');
 
+  if (options.config) { // update paths to config path
+    config.tests = path.resolve(testRoot, config.tests);
+  }
+
 
   const childProcessesPromise = new Promise((resolve, reject) => {
     processesDone = resolve;

--- a/lib/command/run-multiple/collection.js
+++ b/lib/command/run-multiple/collection.js
@@ -98,6 +98,15 @@ class Collection {
       const runConfig = run.getConfig();
       const browsers = [];
 
+      if (!Array.isArray(runConfig.browsers)) {
+        runConfig.browsers = guessBrowser(config) || [];
+      }
+
+      // no browser property in helper?
+      if (!runConfig.browsers.length) {
+        runConfig.browsers.push('default');
+      }
+
       runConfig.browsers.forEach((browser) => {
         const browserConfig = browser.browser ? browser : { browser };
         const runBrowserConfig = { ...runConfig, browser: browserConfig };
@@ -152,6 +161,15 @@ class Collection {
 const createCollection = (runs) => {
   return new Collection(runs);
 };
+
+function guessBrowser(config) {
+  const firstHelper = Object.keys(config.helpers)[0];
+  if (!firstHelper) throw new Error('No helpers declared!');
+  if (!config.helpers[firstHelper].browser) {
+    return [];
+  }
+  return [config.helpers[firstHelper].browser];
+}
 
 module.exports = {
   createRuns,

--- a/lib/command/run.js
+++ b/lib/command/run.js
@@ -2,7 +2,6 @@ const getConfig = require('./utils').getConfig;
 const getTestRoot = require('./utils').getTestRoot;
 const deepMerge = require('./utils').deepMerge;
 const fileExists = require('../utils').fileExists;
-const detectAbsolutePath = require('../utils').detectAbsolutePath;
 const path = require('path');
 const mkdirp = require('mkdirp');
 const Config = require('../config');
@@ -23,7 +22,7 @@ module.exports = function (test, options) {
     config = Config.append(JSON.parse(options.override));
   }
 
-  if (detectAbsolutePath(config.output)) outputDir = config.output;
+  if (path.isAbsolute(config.output)) outputDir = config.output;
   else outputDir = path.join(testRoot, config.output);
 
   if (!fileExists(outputDir)) {

--- a/lib/utils.js
+++ b/lib/utils.js
@@ -52,12 +52,6 @@ module.exports.fileExists = function (filePath) {
   return true;
 };
 
-module.exports.detectAbsolutePath = function (filePath) {
-  if (filePath.startsWith('/') || filePath.startsWith('\\')) return true;
-  if (filePath.match(/^\w\:\\/)) return true; // Windows style
-  return false;
-};
-
 module.exports.isFile = function (filePath) {
   let filestat;
   try {

--- a/test/data/sandbox/codecept.multiple.json
+++ b/test/data/sandbox/codecept.multiple.json
@@ -13,7 +13,7 @@
   "multiple": {
     "default": {
       "browsers": [
-        "chrome", 
+        "chrome",
         { "browser": "firefox"}
       ]
     },
@@ -38,6 +38,9 @@
         "chrome",
         { "browser": "firefox", "windowSize": "1200x840"}
       ]
+    },
+    "chunks": {
+      "chunks": 2
     }
   },
   "include": {},

--- a/test/runner/run_multiple_test.js
+++ b/test/runner/run_multiple_test.js
@@ -112,4 +112,15 @@ describe('CodeceptJS Multiple Runner', function () {
       done();
     });
   });
+
+  it('should run chunks', (done) => {
+    exec(`${codecept_run}chunks`, (err, stdout, stderr) => {
+      stdout.should.include('CodeceptJS'); // feature
+      stdout.should.include('[1.chunks:chunk1:dummy] print browser');
+      stdout.should.include('[2.chunks:chunk2:dummy] @grep print browser size');
+      assert(!err);
+      done();
+    });
+  });
 });
+

